File: test-automation-core/src/main/java/com/nortal/test/core/rest/error/SwaggerValidationException.java
package com.nortal.test.core.rest.error;

import java.util.List;
import com.atlassian.oai.validator.report.ValidationReport.Message;

/**
 * Exception representing a swagger validation failure.
 */
public class SwaggerValidationException extends ValidationException {
	private static final long serialVersionUID = 924828565647941758L;

	protected static String messagesToString(List<Message> messages) {
		StringBuilder builder = new StringBuilder("Validation failed:");
		for (Message message : messages) {
			builder.append('\n');
			builder.append(message.toString());
		}
		return builder.toString();
	}

	protected final List<Message> messages;

	public SwaggerValidationException(List<Message> messages) {
		super(messagesToString(messages));
		this.messages = messages;
	}

	public List<Message> getMessages() {
		return messages;
	}
}

File: test-automation-core-old/src/main/java/com/nortal/test/core/rest/error/SwaggerValidationException.kt
package com.nortal.test.core.rest.error

import com.atlassian.oai.validator.report.ValidationReport
import java.lang.StringBuilder

/**
 * Exception representing a swagger validation failure.
 */
class SwaggerValidationException(val messages: List<ValidationReport.Message>) : ValidationException(
    messagesToString(
        messages
    )
) {

    companion object {
        private const val serialVersionUID = 924828565647941758L
        protected fun messagesToString(messages: List<ValidationReport.Message>): String {
            val builder = StringBuilder("Validation failed:")
            for (message in messages) {
                builder.append('\n')
                builder.append(message.toString())
            }
            return builder.toString()
        }
    }
}